src/AspNet.Security.OAuth.Reddit/RedditAuthenticationHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Framework.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Reddit {
+
+    public class RedditAuthenticationHandler : OAuthAuthenticationHandler<RedditAuthenticationOptions> {
+
+        public RedditAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("bearer", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, RedditAuthenticationHelper.GetIdentifier(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim(ClaimTypes.Name, RedditAuthenticationHelper.GetName(payload), Options.ClaimsIssuer)
+                     .AddOptionalClaim("urn:Reddit:Name", RedditAuthenticationHelper.GetName(payload), Options.ClaimsIssuer)
+                     .AddOptionalClaim("urn:Reddit:Over_18", RedditAuthenticationHelper.GetOver18(payload), Options.ClaimsIssuer);
+
+            var context = new OAuthAuthenticatedContext(Context, Options, Backchannel, tokens, payload) {
+                Principal = new ClaimsPrincipal(identity),
+                Properties = properties
+            };
+
+            await Options.Events.Authenticated(context);
+
+            if (context.Principal?.Identity == null) {
+                return null;
+            }
+
+            return new AuthenticationTicket(context.Principal, context.Properties, Options.AuthenticationScheme);
+        }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri) {
+            return $"{base.BuildChallengeUrl(properties, redirectUri)}&duration=permanent";
+        }
+
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] string code, [NotNull] string redirectUri) {
+            var credentials = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{Options.ClientId}:{Options.ClientSecret}"));
```

Comment:
FYI, I'll revert this change in the Yahoo provider to avoid weird localization issues in the future (string interpolation is culture-dependent by default).